internal/platform/gin/auditlog/auditlogdriver/db_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auditlogdriver
+
+import (
+	"testing"
+	"time"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/sqlite" // SQLite driver used for integration test
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/banzaicloud/pipeline/internal/common"
+	"github.com/banzaicloud/pipeline/internal/platform/gin/auditlog"
+)
+
+func setUpDatabase(t *testing.T) *gorm.DB {
+	db, err := gorm.Open("sqlite3", "file::memory:")
+	require.NoError(t, err)
+
+	err = Migrate(db, common.NoopLogger{})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		_ = db.Close()
+	})
+
+	return db
+}
+
+func TestDatabaseDriver(t *testing.T) {
+	entry := auditlog.Entry{
+		Time:          time.Date(1984, time.April, 4, 0, 0, 0, 0, time.UTC),
```

Comment:
**Question**: just curious, whose birth date is this? 🙂

Reply:
Hint:

> To the past, or to the future. To an age when thought is free. From the Age of Big Brother, from the Age of the Thought Police, from a dead man - greetings!

😉